pagopa/io-app
# fix: [IOCOM-1556] Fix missing message list on Android, new messages'home

## Short description
This PR fixes a regression in the new messages' home on Android, where the messages' list had disappeared.

|Pull to refresh<br/>iOS|Pull to refresh<br/>Android|
|-|-|
|<video src="https://github.com/pagopa/io-app/assets/5150343/eec7fcbf-6fb7-4224-bf91-614177d6e1c0">|<video src="https://github.com/pagopa/io-app/assets/5150343/5e6f9a6b-49a4-45a7-808d-4d97f1b4a332">|

## List of changes proposed in this pull request
- On Android, a RefreshControl wrapped inside a React Component results in a broken message list. CustomRefreshControl has been deleted and its logic has been moved to the MessageList (causing additional vDOM re-renderings when the refresh status changes)

## How to test
Using the io-dev-api-server, make sure to enable both DS and new messages' home. Pull to refresh.
